ramalama/sandbox.py

```
@@ -296,10 +331,52 @@ def run_sandbox(args: SandboxEngineArgsType, agent_cls: type[Agent]):
+
+def _query_router_models(port) -> list[str]:
+    """Query the llama.cpp router server for available model IDs."""
+    try:
+        conn = HTTPConnection("127.0.0.1", int(port), timeout=5)
+        conn.request("GET", "/v1/models")
+        resp = conn.getresponse()
+        if resp.status != 200:
+            return []
+        body = json.loads(resp.read())
+        return [m.get("id", "") for m in body.get("data", []) if m.get("id")]
+    except Exception:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)